numpy/distutils/exec_command.py

```diff
@@ -249,14 +249,25 @@ def _exec_command(command, use_shell=None, use_tee = None, **env):
         proc = subprocess.Popen(command, shell=use_shell, env=env,
                                 stdout=subprocess.PIPE,
                                 stderr=subprocess.STDOUT,
-                                universal_newlines=True)
+                                universal_newlines=False)
     except EnvironmentError:
         # Return 127, as os.spawn*() and /bin/sh do
         return 127, ''
+
     text, err = proc.communicate()
+    if sys.version_info >= (3, 0):
+        text = text.decode(sys.getdefaultencoding(), errors='replace')
+
+    text = text.replace('\r\n', '\n')
     # Another historical oddity
     if text[-1:] == '\n':
         text = text[:-1]
+
+    # Python 2.7 doesn't distinguish between text and bytes,
+    # so to avoid issues, we simply remove all non-ascii characters
+    if sys.version_info < (3, 0):
+        text = ''.join(i for i in text if ord(i) < 128)
```

Comment:
`text.encode('ascii', errors='replace')` ?

Reply:
Tried; did not work.